pkg/api/store/workload/workload_store.go

```diff
@@ -327,3 +329,22 @@ func getDomain(image string) string {
 	}
 	return domain
 }
+
+func imageUpdated(apiContext *types.APIContext, data map[string]interface{}) (bool, error) {
+	var workload projectclient.Workload
+	accessError := access.ByID(apiContext, &projectschema.Version, "workload", apiContext.ID, &workload)
+	if accessError != nil {
+		return false, httperror.NewAPIError(httperror.InvalidReference, "Error accessing workload")
+	}
+
+	containers, _ := values.GetSlice(data, "containers")
+	if len(containers) != len(workload.Containers) {
+		return true, nil
+	}
+	for i := range containers {
```

Comment:
we should account for the fact that number of containers can change. Like it can be 1 for existing workload, and 2 in update, and vice versa. In case when len(containers1) != len(containers2), we can return true right away.

Reply:
https://github.com/rancher/rancher/pull/14240#discussion-diff-198655249R341